tests/pipelines/test_pipelines.py

```diff
@@ -1802,6 +1802,17 @@ def test_pipe_same_device_id_offload(self):
         sd.maybe_free_model_hooks()
         assert sd._offload_gpu_id == 5
 
+    def test_wrong_model(self):
+        tokenizer = CLIPTokenizer.from_pretrained("hf-internal-testing/tiny-random-clip")
+        with self.assertRaises(ValueError) as error_context:
+            _ = StableDiffusionPipeline.from_pretrained(
+                "hf-internal-testing/diffusers-stable-diffusion-tiny-all", text_encoder=tokenizer
+            )
+
+        assert "Expected" in str(error_context.exception)
+        assert "text_encoder" in str(error_context.exception)
+        assert f"{tokenizer.__class__.__name}" in str(error_context.exception)
```

Comment:
Maybe also a check for the `scheduler` as that is handled slightly differently?

Reply:
Will add it. For context this is what we're handling:

https://github.com/huggingface/diffusers/blob/d041dd504058ac6b0fde3eb767eb6844d8d577b8/src/diffusers/schedulers/scheduling_utils.py#L33-L48

Reply:
That's cool. But I don't see the flow matching schedulers here. So, if I do assign a text encoder to `scheduler` in an RF pipeline (`FluxPipeline`, for example), would it still work as expected?

Reply:
Yes that also works, for pipelines like Flux we're getting the `type`
```python
<class 'diffusers.schedulers.scheduling_flow_match_euler_discrete.FlowMatchEulerDiscreteScheduler'>
```
For SD etc we get the `enum`
```python
<enum 'KarrasDiffusionSchedulers'>
```
```python
[<KarrasDiffusionSchedulers.DDIMScheduler: 1>, <KarrasDiffusionSchedulers.DDPMScheduler: 2>, <KarrasDiffusionSchedulers.PNDMScheduler: 3>, <KarrasDiffusionSchedulers.LMSDiscreteScheduler: 4>, <KarrasDiffusionSchedulers.EulerDiscreteScheduler: 5>, <KarrasDiffusionSchedulers.HeunDiscreteScheduler: 6>, <KarrasDiffusionSchedulers.EulerAncestralDiscreteScheduler: 7>, <KarrasDiffusionSchedulers.DPMSolverMultistepScheduler: 8>, <KarrasDiffusionSchedulers.DPMSolverSinglestepScheduler: 9>, <KarrasDiffusionSchedulers.KDPM2DiscreteScheduler: 10>, <KarrasDiffusionSchedulers.KDPM2AncestralDiscreteScheduler: 11>, <KarrasDiffusionSchedulers.DEISMultistepScheduler: 12>, <KarrasDiffusionSchedulers.UniPCMultistepScheduler: 13>, <KarrasDiffusionSchedulers.DPMSolverSDEScheduler: 14>, <KarrasDiffusionSchedulers.EDMEulerScheduler: 15>]
```
So we apply the same processing (`str`, `split`, `strip` applies for `type` case) to get a list of scheduler
```python
['FlowMatchEulerDiscreteScheduler']
```
```python
['DDIMScheduler', 'DDPMScheduler', 'PNDMScheduler', 'LMSDiscreteScheduler', 'EulerDiscreteScheduler', 'HeunDiscreteScheduler', 'EulerAncestralDiscreteScheduler', 'DPMSolverMultistepScheduler', 'DPMSolverSinglestepScheduler', 'KDPM2DiscreteScheduler', 'KDPM2AncestralDiscreteScheduler', 'DEISMultistepScheduler', 'UniPCMultistepScheduler', 'DPMSolverSDEScheduler', 'EDMEulerScheduler']
```
If it's not a scheduler it will raise or if it's the wrong type of scheduler.

Reply:
We now also support `Union`

Reply:
Tests for wrong scheduler are added.